File: Semester_3_Angular_Personal/Admin/src/app/services/account.service.ts
import { Injectable } from "@angular/core";
import { BaseURLService } from "./baseUrl.service";
import { HttpClient } from "@angular/common/http";
import { Login } from "../models/login.model";
import { lastValueFrom } from "rxjs";
import { Account } from "../models/account.model";

@Injectable()
export class AccountService {
    constructor(
        private baseURlService: BaseURLService,
        private httpClient: HttpClient,
    ){}

    async login(login: Login){
        return await lastValueFrom(this.httpClient.post(this.baseURlService.baseURL() + 'admin/account/login', login));
    }

    async findAll(){
        return await lastValueFrom(this.httpClient.get(this.baseURlService.baseURL() + 'admin/account/findAll'));
    }

    async findByEmail(email: string){
        return await lastValueFrom(this.httpClient.get(this.baseURlService.baseURL() + 'admin/account/findByEmail/' + email));
    }

    async create(account: Account) {
        return await lastValueFrom(this.httpClient.post(this.baseURlService.baseURL() + 'admin/account/register', account));
    }

    async edit(account: Account){
        return await lastValueFrom(this.httpClient.put(this.baseURlService.baseURL() + 'admin/account/edit', account));
    }

    async find(id: string) {
        return await lastValueFrom(this.httpClient.get(this.baseURlService.baseURL() + 'admin/account/find/' + id));
    }

    async upload(id: string, file: File) {
        var formData = new FormData();
        formData.append('file', file);
        return await lastValueFrom(this.httpClient.post(this.baseURlService.baseURL() + 'admin/account/uploadFile/' + id, formData));
    }
}

File: Semester_3_API_Personal/Semester_3_API_Personal/Service/AccountService.cs

using Semester_3_API_Personal.Models;

namespace Semester_3_API_Personal.Service;


public interface AccountService
{
    public bool login(string email, string password);
    public bool login2(string email, string password);
    public bool register(Account account);
    public bool edit(Account account);
    public bool Delete(int id);
    public dynamic findAll();
    public dynamic find(int id);
    public Account findByIdNoTracking(int id);
    public dynamic findbyEmail(string email);
    public dynamic findbyStatus(bool status);
    // Check Email
    public bool CheckMail(string username);
    public dynamic VerifyCode(string account);
    public bool Active(string email, string securityCode);
}
